WordPress/Classes/ViewRelated/Me/All Domains/View Models/AllDomainsListViewModel.swift

```diff
@@ -59,18 +62,53 @@ class AllDomainsListViewModel {
         return .init(domain: domains[index])
     }
 
+    // MARK: - Creating Empty State View Models
+
+    /// The empty state to display when the user doesn't have any domains.
+    private func emptyStateViewModel() -> AllDomainsListEmptyStateViewModel {
+        return .init(
+            title: Strings.emptyStateTitle,
+            description: Strings.emptyStateDescription,
+            button: .init(title: Strings.emptyStateButtonTitle, action: {})
+        )
+    }
+
+    /// The empty state to display when an error occurs.
+    private func emptyStateViewModel(from error: Error) -> AllDomainsListEmptyStateViewModel {
+        let title: String
+        let description: String
+        let button: AllDomainsListEmptyStateViewModel.Button = .init(title: Strings.errorStateButtonTitle) { [weak self] in
+            self?.loadData()
+        }
+
+        let nsError = error as NSError
+        if nsError.domain == NSURLErrorDomain, nsError.code == NSURLErrorNotConnectedToInternet {
+            title = Strings.offlineEmptyStateTitle
+            description = Strings.offlineEmptyStateDescription
+        } else {
+            title = Strings.errorEmptyStateTitle
+            description = Strings.errorEmptyStateDescription
+        }
+
+        return .init(title: title, description: description, button: button)
+    }
+
+    /// The empty state to display when there are no domains matching the search query.
+    private func emptyStateViewModel(searchQuery: String) -> AllDomainsListEmptyStateViewModel {
+        fatalError("Not implemented yet")
+    }
```

Comment:
This will be implemented in a future PR as part of the All Domains Search feature https://github.com/wordpress-mobile/WordPress-iOS/issues/21560